### ISSUE TITLE
Audit nteract notebook/runtime pronunciation overrides

### ISSUE THREAD

**Issue by quillaid** [rgbkrk/voice#90]
## Context

While preparing #89, I mined the local `nteract/nteract` checkout at `/home/ubuntu/projects/nteract` for terms Hermes/voice is likely to read aloud. The scan excluded `node_modules`, `target`, generated bundles, coverage, lockfiles, maps, minified assets, and image assets.

#89 covers the broadly useful developer terms that overlapped with nteract usage: JSON, URL/URI, WASM, Serde, Vite/Vitest, MCP, Tokio, Jupyter, UUID, ipynb, CRDT, AWS, pnpm, YAML/TOML, SQLite/Postgres, and WebRTC.

## Follow-on pronunciation candidates

Notebook/runtime nouns:

- `runtimed`
- `Automerge`
- `RuntimeStateDoc`
- `NotebookDoc`
- `CommsDoc`
- `kernelspec`
- `ipykernel`
- `nbformat`
- `nbconvert`

Protocol/auth terms:

- `OIDC`
- `MIME`
- `MIMEBundle`
- `WebSocket`
- `ZMQ`
- `ZeroMQ`
- `S3`
- `ULID`
- `OPFS`
- `PWA`
- `SSR`
- `CSR`
- `DOM`
- `JSX`
- `TSX`
- `gRPC`
- `Protobuf`

Renderer/data terms:

- `Arrow`
- `Parquet`
- `DataFrame`
- `DuckDB`
- `Polars`
- `Plotly`
- `Vega`
- `Vega-Lite`
- `Leaflet`
- `CodeMirror`
- `KaTeX`
- `MathJax`
- `LaTeX`
- `TeX`
- `MDX`
- `Pyodide`
- `micropip`

App/tooling names:

- `Tauri`
- `Cloudflare`
- `Wrangler`
- `Playwright`
- `Yjs`
- `Specta`
- `ts-rs`
- `Monaco`

## Suggested approach

Use the same pattern as #89: add high-confidence single-token overrides to `voice-g2p`, keep phrase substitutions in the CLI only when they cannot be represented as a single lookup key, and add focused assertions to `test_builtin_overrides`.

**Comment by quillaid** [comment]
Follow-on PR opened: #91.

#91 covers the high-confidence terms from this audit that are normal single-token G2P override candidates: `runtimed`, `Automerge`, `kernelspec`, `ipykernel`, `nbformat`, `nbconvert`, `OIDC`, `MIMEBundle`, `WebSocket`, `ZMQ`, `ZeroMQ`, `S3`, `ULID`, `OPFS`, `PWA`, `SSR`, `CSR`, `JSX`, `TSX`, `gRPC`, `Protobuf`, `DuckDB`, `KaTeX`, `MathJax`, `LaTeX`, `TeX`, `MDX`, `Pyodide`, `micropip`, `Tauri`, `Cloudflare`, and `Yjs`.

Not added in #91:

- Terms that already probed acceptably with the current G2P output: `RuntimeStateDoc`, `NotebookDoc`, `CommsDoc`, `MIME`, `Arrow`, `Parquet`, `DataFrame`, `Polars`, `Plotly`, `Vega`, `Vega-Lite`, `Leaflet`, `CodeMirror`, `Wrangler`, `Playwright`, `Specta`, `Monaco`.
- `ts-rs`, because hyphenated lowercase chunks do not currently hit the whole-token override path. That should be handled as tokenizer/phrase override behavior rather than squeezed into a pronunciation-data patch.

**Comment by quillaid** [comment]
Added another commit to #91 for terms mined from the voice repo and the local Hermes command-provider setup.

New voice/audio terms covered there: `STFT`, `RTP`, `RTCP`, `PCM`, `FLAC`, `OGG`, `JSONRPC`, `PyO3`, `LSTM`, `BiLSTM`, `MLX`, `GGUF`, `CoreAudio`, `CoreML`, `tokenizer`, `tokenizers`, `mmap`, and `espeak`.

Terms probed but left alone because current output was acceptable or the expected pronunciation was less clear: `Whisper`, `distil-whisper`, `Candle`, `Metal`, `HuggingFace`, `Hermes`, `WhatsApp`, `Opus`, `ALBERT`, `Misaki`, `grapheme`, `phoneme`, `vocoder`, `daemon`, `latency`, `jitter`, `PulseAudio`, and `PipeWire`. Ambiguous audio-library names such as `CPAL`, `WASAPI`, and `ALSA` should get a separate decision if we care about those enough to pin them.

**Comment by quillaid** [comment]
Added a third follow-on commit to #91 from the local Hermes/provider scan.

New Hermes/provider terms covered there: `GPT`, `ChatGPT`, `Neuphonic`, `NeuTTS`, `Node.js`, `NodeJS`, `QQBot`, and `xAI`.

Terms probed but left alone because the current output was acceptable or the desired pronunciation was less certain: `OpenAI`, `Codex`, `Hermes`, `Nous`, `Mistral`, `Mixtral`, `Qwen`, `DeepSeek`, `Llama`, `Ollama`, `Gemini`, `Google`, `Vertex`, `Groq`, `Grok`, `Perplexity`, `HuggingFace`, `Voxtral`, `ElevenLabs`, `Azure`, `Aria`, `Alloy`, `WhatsApp`, `Telegram`, `Discord`, `Slack`, `Signal`, `HomeAssistant`, `Yuanbao`, `Teams`, `GoogleChat`, `Sage`, `Router`, `Daytona`, `Modal`, `Singularity`, `Docker`, `Anaconda`, `Pixi`, `uv`, `conda`, and `OpenRouter`.

**Comment by quillaid** [comment]
Added a fourth follow-on commit to #91 from the notebook data-science/Python tooling scan.

New terms covered there: `Matplotlib`, `SciPy`, `scikit-learn`, `sklearn`, `PyArrow`, `IPython`, `ipywidgets`, `anywidget`, `PyPI`, `mypy`, `pytest`, `isort`, and `JAX`.

Terms probed but left alone because current output was acceptable or the desired pronunciation was less clear: `pandas`, `DataFrame`, `DataFrames`, `Polars`, `Plotly`, `PlotlyJS`, `Vega`, `VegaLite`, `Vega-Lite`, `Arrow`, `Parquet`, `Numba`, `TensorFlow`, `Keras`, `XArray`, `Dask`, `Ray`, `JupyterLab`, `Binder`, `Voila`, `Papermill`, `conda`, `mamba`, `micromamba`, `Pixi`, `uv`, `pip`, `poetry`, `hatch`, `ruff`, and `black`.

**Comment by quillaid** [comment]
Added a fifth follow-on commit to #91 from the local frontend/browser/build-tooling scan.

New terms covered there: `CSS`, `SCSS`, `HTML`, `Next.js`, `NextJS`, `TailwindCSS`, `iframe`, `IndexedDB`, `IDB`, `esbuild`, `SWC`, `ESLint`, `MSW`, and `Vue`.

Terms probed but left alone because current output was acceptable or pronunciation is less settled: `React`, `ReactDOM`, `Preact`, `Solid`, `Svelte`, `Vite`, `Vitest`, `pnpm`, `npm`, `npx`, `yarn`, `bun`, `Deno`, `Tauri`, `Electron`, `Radix`, `RadixUI`, `radix-ui`, `Tailwind`, `Lucide`, `shadcn`, `TanStack`, `Zustand`, `Redux`, `RxJS`, `CodeMirror`, `Monaco`, `markdown`, `frontmatter`, `Sass`, `DOM`, `ShadowDOM`, `WebView`, `WebWorker`, `ServiceWorker`, `localStorage`, `Rollup`, `Webpack`, `Babel`, `Prettier`, `TypeScript`, `JavaScript`, `Playwright`, `Cypress`, `Storybook`, `Chromium`, `WebKit`, and `Firefox`.

**Comment by quillaid** [comment]
Follow-on audit from the current `nteract/nteract` wording after #91.

I fast-forwarded the local checkout at `/home/ubuntu/projects/nteract` to `origin/main` and re-scanned current docs/source. The main remaining pronunciation gap is not another broad list of normal words; it is identifier-shaped technical wording that appears frequently in nteract text.

Representative current nteract wording:

- `README.md` names `nteract`, `runtimed`, `runt`, Automerge, and MCP in the product/runtime overview.
- `docs/audits/runtime-peer-and-blob-authority-audit.md` repeatedly names `runtime_peer`, room `ACL`, `NotebookDoc`, `RuntimeStateDoc`, `CommsDoc`, `JupyterHub`, and Cloudflare DO topology.
- `src/components/widgets/AGENTS.md` names `RuntimeStateDoc`, `CommsDoc`, `SyncEngine`, `CommBridgeManager`, `JSON-RPC`, `bufferPaths`, `DataView`, `ContentRef`, `ipywidgets`, and `nteract.dx.*`.
- renderer/Sift wording names `Arrow IPC`, `Parquet`, `GeoJSON`, `Leaflet`, `Vega`, `Plotly`, and `Sift`.

Counts from the targeted scan, excluding generated/lock/build noise where practical:

- `RuntimeStateDoc`: 1364
- `NotebookDoc`: 836
- `runtime_peer`: 528
- `CommsDoc`: 439
- `ACL`: 301
- `Arrow IPC`: 195
- `SyncEngine`: 113
- `JupyterHub`: 107
- `D1`: 99
- `RuntimeStateSync`: 94
- `PutBlob`: 81
- `R2`: 79
- `Cloudflare`: 72
- `AutomergeSync`: 70
- `PUT_BLOB`: 63
- `CommBridgeManager`: 48
- `automunge`: 39
- `Outerbounds`: 39
- `nteract-dev`: 36
- `Utf8View`: 24
- `outputIdChanges`: 17
- `repr-llm`: 16
- `Wrangler`: 16

Focused G2P probes on #91 show these are the notable misses:

```text
ACL -> ˌAsˌiˈɛl
Arrow IPC -> ˈɛɹO ˌIpˌisˈi
BYOC -> bˌiwˌIˌOsˈi
automunge -> ˈæʌtɑmʌŋ
Outerbounds -> ˈWtɜɹbWnds
Utf8View -> jˈu tˈi ˈɛf ˈAt vjˈu
outputIdChanges -> ˈWtpˌʊt ˌɪd ʧˈAnʤᵻz
```

Several CamelCase identifiers are tolerable but should still be covered by tests if we plan to read nteract docs/logs aloud often:

```text
RuntimeStateDoc -> ɹˈʌntIm stˈAt dˌɑk
NotebookDoc -> nˈOtbˌʊk dˌɑk
CommsDoc -> kˈɑmz dˌɑk
SyncEngine -> sˌɪŋk ˈɛnʤən
RuntimeStateSync -> ɹˈʌntIm stˈAt sˌɪŋk
AutomergeSync -> ˈɔTO mˈɜɹʤ sˌɪŋk
CommBridgeManager -> kˌɑm bɹˈɪʤ mˈænɪʤəɹ
```

Recommended next PR after #91: **identifier-aware pronunciation inspection and nteract corpus coverage**.

Scope I would keep in that PR:

1. Add `voice phonemes` as a first-class CLI inspection command so pronunciation audits do not require a temporary probe binary or TTS side effects.
2. Add a small nteract-derived regression corpus in `voice-g2p` tests using representative phrases, not just isolated terms.
3. Add high-confidence missing overrides for recurring acronyms/proper nouns: `ACL`, `IPC`, `BYOC`, `Outerbounds`, `Utf8View`, and likely `automunge`.
4. Improve identifier normalization for common docs/code forms before falling back to per-word G2P: snake_case (`runtime_peer`), SCREAMING_SNAKE (`PUT_BLOB`), kebab/package names (`runt-nightly`, `runtimed-wasm`, `nteract-dev`, `repr-llm`), CamelCase (`RuntimeStateDoc`, `CommBridgeManager`, `outputIdChanges`), and dotted namespaces (`nteract.dx.*`).
5. Keep `D1`, `R2`, `runt`, `runtimed`, `Sift`, `Wrangler`, `Miniflare`, and renderer names in the corpus as guardrails even where current output is acceptable.

This should be a separate PR from #91 because it changes pronunciation mechanics and audit ergonomics, not just override data.

**Comment by quillaid** [comment]
Corrected follow-on note: opened PR #92 for the identifier-aware pass from the latest audit: https://github.com/rgbkrk/voice/pull/92

This adds `voice phonemes` for direct G2P inspection, pins missing nteract identifier/acronym overrides (`ACL`, `IPC`, `BYOC`, `automunge`, `Outerbounds`, `Utf8View`, `outputIdChanges`, `PUT_BLOB`, `runtime_peer`, `runtimed-wasm`, etc.), treats dotted namespace separators as word boundaries, and adds phrase-level nteract regression coverage for `RuntimeStateDoc`, `CommsDoc`, `SyncEngine`, `CommBridgeManager`, `nteract.dx.blob`, `Arrow IPC`, and related wording.

It is stacked on #91, which is still stacked on #89.

**Comment by quillaid** [comment]
Follow-up residual audit after rebasing #91/#92 found one more common developer-text pattern worth fixing in #92: initialisms like `ID`, `IDs`, `UI`, `UX`, and `JS` were still being pronounced as glued words in standalone and identifier contexts, and acronym prefixes inside identifiers were not split/reused correctly (for example `HTMLElement` and `URLSearchParams`).\n\nI amended #92 to add those direct overrides, let grouped identifier resolution consult builtin overrides for subspans before the lexicon, and split acronym prefixes before capitalized identifier words. New regression coverage now checks `HTMLElement`, `URLSearchParams`, `cell_id`, and `NotebookUI` phrase behavior, plus exact standalone override assertions.

**Comment by quillaid** [comment]
One more residual browser/UI probe found `HTMLIFrameElement` was frequent enough to cover directly: the generic acronym split reasonably read it as `HTMLI` + `Frame`, which glues the `HTML` letters into the iframe class name. I amended #92 with a direct `HTMLIFrameElement` override and exact regression coverage rather than making the tokenizer guess that framework-specific split globally.

**Comment by quillaid** [comment]
Additional WebRTC/Hermes streaming vocabulary probe from `docs/streaming.md`, daemon stream code, and the local Hermes command-provider config found five clear misses now added to #91: `SDP`, `DTLS`, `SRTP`, `CPAL`, and `Rodio`.

Other checked terms either already had good direct overrides or read acceptably without one: `Hermes`, `WhatsApp`, `WebRTC`, `WebSocket`, `Opus`, `OGG`, `PCM`, `WAV`, `TTS`, `STT`, `RTP`, `RTCP`, `STUN`, `TURN`, `ICE`, `SIP`, `queue_id`, `stream_id`, `sample_rate`, `frame_ms`, and `client_id`.

#92 was rebased on the updated #91 stack and revalidated locally.

**Comment by quillaid** [comment]
Another cleaned source/config probe focused on model and tool names found four clear misses now added to #91: `Kokoro-82M`, `DFT`, `LFS`, and `openai-codex`.

I left the following probed terms without new direct overrides because they already read acceptably or are too code-internal for this pronunciation-data pass: `HuggingFace`, `distil-whisper`, `distil-large-v3`, `Candle`, `SafeTensors`, `VarBuilder`, `DType`, `DeviceSinkBuilder`, `SamplesBuffer`, `SampleFormat`, `UnixStream`, `NonZero`, `SeqCst`, `AdaIN`, `AdaINResBlk1d`, `ResBlk1d`, `OpenAI`, `Anthropic`, `Claude`, `Gemini`, `OpenRouter`, `Voxtral`, `NeuTTS`, `scribe_v2`, `image_gen`, `tool_search`, `code_execution`, `computer_use`, `HomeAssistant`, and `GoogleChat`.

#92 was rebased on the updated #91 stack and revalidated locally.

**Comment by rgbkrk** [comment]
This is done. Agent went a little extra. Thanks buddy.
